sys/sys/mbuf.h

```diff
@@ -924,6 +925,7 @@ m_extaddref(struct mbuf *m, char *buf, u_int size, u_int *ref_cnt,
 
 	atomic_add_int(ref_cnt, 1);
 	m->m_flags |= M_EXT;
+	/* XXX-AM: Should we have an option to warn when these are not exact? */
```

Comment:
These can also be arbitrary buffers.  I'm not sure the caller is going to be able to do anything though if it is not exact.  For example, the cxgbe(4) driver uses this for a feature it calls buffer packing where it allocates a large (multi-page) receive buffer and the NIC packs multiple received packets into the same large buffer.  The driver allocates mbufs that then use the large buffer as the backing store but only reference the subset of the buffer.  You can certainly use this with 9k jumbo frames where the resulting size passed in `size` might not be exactly representable?

Reply:
I wonder if what we actually want is a counter for inexact bounds?

Reply:
I wonder whether we want one specific for mbufs or a general one. I provisionally added a counter specifically for inexact ext_buf bounds.